scripts/cli/TestRunnerOfflineMode.php

```diff
@@ -0,0 +1,271 @@
+    }
+
+    /**
+     * Turning on offline mode for test runner. It sets special proxy plugin that supports offline actions,
+     * as well it deactivates plugins that unsupported in offline mode. Also, it adjusts test runner configuration.
+     *
+     * @return Report
+     * @throws InvalidServiceManagerException
+     * @throws common_exception_Error
+     * @throws common_exception_InconsistentData
+     * @throws common_ext_ExtensionException
+     */
+    private function enable()
+    {
+        $this->updateTestRunnerConfig(
+            [
+                'allow-browse-next-item' => true,
+                'bootstrap' => [
+                    'serviceExtension' => 'taoQtiTest',
+                    'serviceController' => 'OfflineRunner',
+                    'communication' => [
+                        'enabled' => true,
+                        'type' => 'request',
+                        'extension' => 'taoQtiTest',
+                        'controller' => 'OfflineRunner',
+                        'action' => 'messages',
+                        'syncActions' => [
+                            'move',
+                            'skip',
+                            'storeTraceData',
+                            'timeout',
+                            'exitTest',
+                            'getNextItemData',
+                        ],
+                    ],
+                ],
+            ]
+        );
+
+        if (!$this->registerOfflineProxy()) {
+            return new Report(Report::TYPE_ERROR, 'Unable to register the proxy.');
+        }
+
+        // set unsupported plugins as inactive
+        $this->handleUnsupportedTestRunnerPlugins(false);
+
+        return new Report(Report::TYPE_SUCCESS, 'Offline Test Runner mode was set.');
+    }
+
+    /**
+     * Turning off the offline mode for test runner. It returns configuration to normal state,
+     * as well returns disabled plugins to active state, if they were used.
+     *
+     * @return Report
+     * @throws InvalidServiceManagerException
+     * @throws common_exception_Error
+     * @throws common_exception_InconsistentData
+     * @throws common_ext_ExtensionException
+     */
+    private function disable()
+    {
+        $this->updateTestRunnerConfig(
+            [
+                'bootstrap' => [
+                    'serviceExtension' => 'taoQtiTest',
+                    'serviceController' => 'Runner',
+                    'communication' => [
+                        'type' => 'poll',
+                        'extension' => null,
+                        'controller' => null,
+                        'action' => 'messages',
+                        'syncActions' => [
+                            'move',
+                            'skip',
+                            'storeTraceData',
+                            'timeout',
+                            'exitTest',
+//                            'getNextItemData', // remove
```

Comment:
If it's not part of the `disable` action, shouldn't it be removed from `enable` as well, or the other way around?

Reply:
Yes, sorry, forget about it. I'll remove. Actually enable action set this parameter, but disable should not. But as there is array_merge used, it is enough just to remove this value from the list.